src/commands/src/helpers/dependency-helpers.ts

```diff
@@ -33,12 +34,47 @@ async function createEnvProdFile(repoDir: string, projectName: string, baseUrl:
   }
 }
 
-export async function prepareBuildAndGetDependencies(repoName: string, projectName: string, baseUrl: string) {
+export async function downloadBuildDependencies(buildDependencies: BuildDependencies, repoName: string) {
+  const commands: string[] = [];
+  await Promise.all(
+    buildDependencies.map(async (dependencyInfo) => {
+      try {
```

Comment:
NIT: I think we should narrow this try/catch to cover where in the code an exception might be thrown i.e. we can wrap just the `executeCliCommand` calls with a try catch. This allows us to be more fine-grained in our errors. For example:

```ts
try {
    await executeCliCommand({
       command: 'mkdir',
       args: ['-p', targetDir],
    });
} catch (e) {
  console.error(`Error! Could not create dependency directory: ${targetDir}. Dependency Information: `, dependencyInfo);
  throw e;

}
```